Remove empty pages & use tmp directory for processing

sarmbruster opened the pull request:
This PR adds an optional feature to remove empty pages. This is useful if you scan one sided documents in duplex mode. 

Additionally I needed to upgrade to scanbd 1.5.x due to a segv in 1.4 upon USB connect (on Debian/Raspian) https://bugs.launchpad.net/ubuntu/+source/scanbd/+bug/1500095. 
A side effect of this upgrade is that the script no longer runs with root permissions - instead it's saned system user. This means we no longer can write intermediary files to the system's root folder. Therefore I've changed the scripts to use a temp folder for this.

Prior to merge thit PR please consider some testing on your side.

rocketraman commented on scan:
```diff
@@ -232,15 +247,16 @@ if [ $numscans -gt 0 ]; then
+      mv $TMP_DIR/scan-0*.pdf "$OUTPUT"
     fi
   fi
+  rm -rf $TMP_DIR
```
Lets use a trap to clean this up...  just like for the lockfile cleanup.

sarmbruster replied:
added.

sarmbruster commented:
Originally I've used it on Raspian (the Debian distro for RaspberryPI). This one uses scanbd 1.4.4. When tracing down issues I did observe that the files from scanadf got placed in the root folder ("/") - which is somewhat scary. Using 1.5.1 I get a permission error since access to root folder is no longer possible (due to different user running this).
I'll fix the PR code as requested.

rocketraman commented:
If you could create a "Contributors" section in the README and add yourself to it, that would be great as well.

sarmbruster commented:
Squashed the commits into one and amended your suggested changes (trap and contributors section).
Note: I've also added a config option to specifiy language used by tesseract

rocketraman commented:
@sarmbruster I noticed you changed the white threshold to 99.99. In one test I did, I had 99.8+. In this test I had a page with 3-holes punched, and the holes created some black shadow on the page. I'm fine with leaving it at the higher percentage but wanted to get your thoughts on reducing it to `> 99.8`.

rocketraman commented:
I'm gonna go ahead and merge with 99.8, but please do let me know your thoughts. We can revert if it makes sense.

sarmbruster commented:
@rocketraman I've the treshold since I had a test document only having some contents in the lower third. With the old value it was erroneously classified as empty page so I increased the treshold. 
I guess we need to somewhat narrow down what an appropriate default value is. Maybe we should provide a config option to override it.
... and thanks for merging.

rocketraman commented:
@sarmbruster would 99.8 have worked for your test pages? Yes an option with a good default makes sense, but it's just as hard for users to figure out what the right value should be as it is for is to come up with a good default. So it's likely such an option will rarely be used

sarmbruster commented:
99.8 did work in my case as well.

rocketraman commented:
> 99.8 did work in my case as well.

Great, sounds like a reasonable compromise then! Thank you for your contribution.
